cmake/FindIntel_SYCL.cmake

```diff
@@ -19,7 +19,7 @@ add_library(INTEL_SYCL::Runtime INTERFACE IMPORTED GLOBAL)
 set_target_properties(INTEL_SYCL::Runtime PROPERTIES
     INTERFACE_LINK_LIBRARIES    OpenCL::OpenCL
     INTERFACE_COMPILE_OPTIONS   "-fsycl;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}"
-    INTERFACE_LINK_OPTIONS      "-fsycl;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}")
+    INTERFACE_LINK_OPTIONS      "-fsycl;-fsycl-device-code-split=per_kernel;-fsycl-targets=${INTEL_SYCL_TRIPLE};${INTEL_SYCL_FLAGS}")
```

Comment:
please change per_kernel to per_source as linking per_kernel cause too much memory consumption on vector_* tests.
Linking per source has reasonable time/memory consumption balance.

Reply:
Done.